iOS/DuckDuckGo/DuckPlayer/DuckPlayerNavigationHandler.swift

```diff
@@ -765,25 +819,68 @@ extension DuckPlayerNavigationHandler: DuckPlayerNavigationHandling {
             return .notHandled(.duplicateNavigation)
         }
         
+        // Ensure all media playback is allowed by default
+        self.toggleMediaPlayback(webView, pause: false)
+
         // Check if DuckPlayer feature is enabled
         guard isDuckPlayerFeatureEnabled else {
             return .notHandled(.featureOff)
         }
         
-        guard let url = webView.url, let (videoID, _) = url.youtubeVideoParams else {
+        guard let url = webView.url, let (videoID, _) = url.youtubeVideoParams else {            
+            duckPlayer.dismissPill()
             return .notHandled(.invalidURL)
         }
         
-        guard url.isYoutubeWatch else {
+        guard url.isYoutubeWatch else {            
+            duckPlayer.dismissPill()
             return .notHandled(.isNotYoutubeWatch)
         }
-        
+
+        // Rest of the function remains the same
         guard videoID != lastWatchInYoutubeVideo else {
             lastURLChangeHandling = Date()
             return .handled(.newVideo)
         }
         
         let parameters = getDuckPlayerParameters(url: url)
+            
+        // Present Bottom Sheet (Native entry point) with delay
+        if duckPlayer.settings.mode == .alwaysAsk && duckPlayer.settings.nativeUI {
+            
+            // First phase: try every 0.05s for 1 second
+            Task { @MainActor in
+                let startTime = Date()
+                
+                // Temporarily pause media playback during page transition
+                // The pause is applied repeatedly for 1 second to ensure it takes effect
+                // even if the DOM is changing during early initialization
+                // Once the page has loaded, the JS mutation observer takes care
+                // Of pausing newly added elements. 
+                while Date().timeIntervalSince(startTime) < 1.0 {
+                    self.toggleMediaPlayback(webView, pause: true)
+                    try? await Task.sleep(nanoseconds: 50_000_000)
+                }
+
+            }
+            
+            // If we're not in a Watch main page, hide
+            // the pill.  Youtube adds #fragments to Watch main pages
+            // When presenting settings and preferences
+            if !url.isYoutubeWatchMainPage {
+                duckPlayer.dismissPill()
+            }
+
+            // Present the Pill if needed
+            Task { @MainActor in
+                try? await Task.sleep(nanoseconds: 1_000_000_000)
```

Comment:
Why the sleep?

Reply:
Aesthethics. We want a little time to render the base layout of the page before showing the Pill.  Prod has yet to define what's the time here